src/box/lua/key_def.h

```diff
@@ -38,6 +38,13 @@ extern "C" {
 struct lua_State;
 struct key_def;
 
+/**
+ * Push key_def as a cdata object to a Lua stack, and set finalizer function
+ * lbox_key_def_gc for it.
+ */
+void
+luaT_push_key_def(struct lua_State *L, const struct key_def *key_def);
```

Comment:
It may be unclear that, despite the `const` specifier, this function actually takes the ownership of the given `key_def` object. This makes it quite different from other similar functions like `luaT_pusherror` or `luaT_pushtuple`. Maybe, better call `key_def_dup` on the argument under the hood?

Reply:
This point looks valid for me too. I overlooked it.

Reply:
OTOH we use it internally to push just created key_def object (and it seems logical to have such a function). Hmm.

Reply:
If it's clearer, I can put
```c
    *(struct key_def **) luaL_pushcdata(L, CTID_STRUCT_KEY_DEF_REF) = new_key_def;
    lua_pushcfunction(L, lbox_key_def_gc);
    luaL_setcdatagc(L, -2);
```
back to `luaT_key_def_merge()` and `lbox_key_def_new()`, instead of calling `luaT_push_key_def()` there,
and use `luaT_push_key_def()` with embedded `key_def_dup()` solely in `luaT_add_index_parts_methods()`.

Reply:
Agreed to duplicate the `key_def` argument in `luaT_push_key_def` and introduce the static function `luaT_key_def_nodup` that would take ownership of `key_def` for use in the module implementation.

Reply:
Done